lib/eslint.js

```diff
@@ -908,6 +908,8 @@ module.exports = (function() {
                 }
             }
 
+            ast.hasShebang = !!shebang;
```

Comment:
That's right - sorry I'm not explaining this better. The `SourceCode` instance above takes the given `ast` from parser and generates its own internal `tokenAndCommentStore` from it. Since this occurs before the shebang comment is removed, the `SourceCode` instance's `tokenAndCommentStore` still contains the shebang comment.

I think we should be able to do the shebang comment removal before creating the `SourceCode` instance - will have to figure out how to do it with the few forks of logic that happen there.